mlc_llm/relax_model/gpt_neox.py

```diff
@@ -32,89 +32,33 @@
 
 @dataclass
 class GPTNeoXConfig:  # pylint: disable=too-many-instance-attributes
-    use_parallel_residual: bool
-    hidden_size: int
-    intermediate_size: int
-    num_attention_heads: int
-    num_hidden_layers: int
-    vocab_size: int
-    rotary_pct: float = 0.25
-    rotary_emb_base: int = 10000
-
-    dtype: str = "float32"
-    layer_norm_eps: float = 1e-05
-    max_sequence_length: int = 2048
-
-
-MODEL_CONFIG = {
```

Comment:
And another thing is just to make sure that you double checked that the config from HF repo is correctly loaded and all entries are consistent. Since we have many models here, checking this might consumes a bit time.

Reply:
I did this check for one of the models (I believe it was dolly) and I was able to confirm that the correct config was being loaded (i.e. the local changes were being brought in correctly).

Reply:
@MasterJH5574 Thanks, found a bug (addressed in https://github.com/mlc-ai/mlc-llm/pull/140/commits/57842aa6a030de5d70a6df109e9cb1d0fbf00b45) in the local path loading. But I've confirmed that it works correctly now for vicuna-v1-7b. Here are my steps:
1. Update config
```shell
(sudeep-mlc-llm) diff config.json config-backup.json
9c9
<   "hidden_size": 100,
---
>   "hidden_size": 4096,
13c13
<   "max_sequence_length": 256,
---
>   "max_sequence_length": 2048,
```

2. Print config being read by `llama.py`
```shell
sudeep-mlc-llm) git diff mlc_llm/relax_model/llama.py
diff --git a/mlc_llm/relax_model/llama.py b/mlc_llm/relax_model/llama.py
index a74b31b..1e9b799 100644
--- a/mlc_llm/relax_model/llama.py
+++ b/mlc_llm/relax_model/llama.py
@@ -46,6 +46,9 @@ class LlamaConfig:
         self.tie_word_embeddings = tie_word_embeddings
         self.position_embedding_base = position_embedding_base
         self.kwargs = kwargs
+    
+    def __str__(self):
+        return f"{self.hidden_size=}, {self.max_sequence_length=}"
```

3. Observe output matches what we expect.
```shell
(sudeep-mlc-llm) python build.py --use-cache 0 --target=vulkan --model-path=/home/sudeepag/vicuna
Using model path /home/sudeepag/vicuna
self.hidden_size=100, self.max_sequence_length=256
```

Let me know if it sounds good to you.

Reply:
Do you mean that the hidden size and max seq length loaded from json are 100 and 256? Not sure if I understand correctly - I think they should be 4096 and 2048 (the values in HF config). I’m quite interested about where the 100 and 256 numbers come from. Did I misunderstand anything?

Reply:
I modified the numbers in my local config (at `/home/sudeep/vicuna/config.json` and am demonstrating that my local changes are being picked up as expected by `llama.py`.

Reply:
I see. That makes perfect sense. Thank you!!